Bottom: ffb1d6a3543ebe1cc1e257921545d874a12bce4e
Top:    6bf16374f9b8f6286e09ee73bcfffb12a0c53eab
Author: Pavel Roskin <proski@gnu.org>
Date:   2013-11-04 09:49:18 -0500

Initial commit of the scripts

---

diff --git a/.farmrc b/.farmrc
new file mode 100644
index 0000000..d2b7400
--- /dev/null
+++ b/.farmrc
@@ -0,0 +1,8 @@
+farm=$HOME/src/farm
+tags="v2.6.14 v2.6.15 v2.6.16 v2.6.17 v2.6.18 v2.6.19 v2.6.20 v2.6.21
+v2.6.22 v2.6.23 v2.6.24 v2.6.25 v2.6.26 v2.6.27 v2.6.28 v2.6.29 v2.6.30
+v2.6.31 v2.6.32 v2.6.33 v2.6.34 v2.6.35 v2.6.36 v2.6.37 v2.6.38 v2.6.39
+v3.0 v3.1 v3.2 v3.3 v3.4 v3.5 v3.6 v3.7 v3.8 v3.9 v3.10 v3.11 v3.12
+linux-next/master"
+makeflags=-j11
+export EXTRA_CFLAGS=-Werror-implicit-function-declaration
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..4fe0eb0
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,22 @@
+INSTALL_PATH = $(HOME)/bin
+RCFILE = $(HOME)/.farmrc
+
+SCRIPTS = farm-checkout farm-clean farm-compile farm-config32 farm-config64 \
+	farm-hardlink farm-warnings
+
+INSTALL = install
+
+all:
+
+install:
+	$(INSTALL) -m 755 -d $(INSTALL_PATH)
+	for scr in $(SCRIPTS); do \
+		$(INSTALL) -m 755 $$scr $(INSTALL_PATH); \
+	done
+	if ! test -e $(RCFILE); then \
+		$(INSTALL) -m 644 .farmrc $(RCFILE); \
+	fi
+
+clean:
+
+.PHONY: all install clean
diff --git a/farm-checkout b/farm-checkout
new file mode 100755
index 0000000..0ac1444
--- /dev/null
+++ b/farm-checkout
@@ -0,0 +1,14 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+cd $HOME/src/linux
+mkdir -p $farm
+for tag in $tags; do
+	srctree=$farm/`echo $tag | sed 's,/,-,'`
+	test -d $srctree && continue
+	mkdir $srctree
+	echo "Checking out $srctree"
+	git --work-tree=$srctree checkout $tag -- .
+done
diff --git a/farm-clean b/farm-clean
new file mode 100755
index 0000000..c7aad8c
--- /dev/null
+++ b/farm-clean
@@ -0,0 +1,11 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+for tree in $farm/*; do
+	test -d $tree/Documentation || continue
+	cd $tree
+	echo "Cleaning $tree"
+	make distclean
+done
diff --git a/farm-compile b/farm-compile
new file mode 100755
index 0000000..32196dd
--- /dev/null
+++ b/farm-compile
@@ -0,0 +1,40 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+# MadWifi
+if test -d tools; then
+	target=modules
+fi
+
+# ndiswrapper
+if test -d utils; then
+	target=driver
+fi
+
+for bintree in $farm/*; do
+	arch=
+	if test -r $bintree/configured-i386; then
+		arch=i386
+		ccflags=-m32
+	fi
+	if test -r $bintree/configured-x86_64; then
+		arch=x86_64
+		ccflags=-m64
+	fi
+	test -n "$arch" || continue
+	log=Log-`basename $bintree`
+	rm -f $log $log-ok $log-error
+	make clean KERNELRELEASE=0 >/dev/null
+	echo -n "Checking $bintree "
+	if make $makeflags HOSTCC="gcc -DPATH_MAX=4096" CC="gcc-4.4 $ccflags" \
+	     ARCH=$arch KBUILD=$bintree KERNELPATH=$bintree C=2 \
+	     CF="-D__CHECK_ENDIAN__" $target >$log 2>&1; then
+		mv $log $log-ok
+		echo -e "\e[32mOK\e[0m"
+	else
+		mv $log $log-error
+		echo -e "\e[31mERROR\e[0m"
+	fi
+done
diff --git a/farm-config32 b/farm-config32
new file mode 100755
index 0000000..55c390a
--- /dev/null
+++ b/farm-config32
@@ -0,0 +1,32 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+arch=i386
+ccflags=-m32
+
+for srctree in $farm/*; do
+	test -d $srctree/Documentation || continue
+	bintree=$srctree-$arch
+	rm -rf $bintree
+	rm -rf $bintree-bad
+	mkdir $bintree
+	cd $srctree
+	make $makeflags HOSTCC="gcc -DPATH_MAX=4096" KBUILD_OUTPUT=$bintree \
+		CONFIG_WIRELESS_EXT=y CONFIG_NET_RADIO=y CONFIG_IPW2100=y \
+		CONFIG_CRYPTO=y ARCH=$arch defconfig
+	cd $bintree
+	sed -i -e 's/# CONFIG_NET_RADIO is not set/CONFIG_NET_RADIO=y/' .config
+	sed -i -e 's/# CONFIG_WIRELESS_EXT is not set/CONFIG_WIRELESS_EXT=y/' .config
+	sed -i -e 's/# CONFIG_IPW2100 is not set/CONFIG_IPW2100=y/' .config
+	sed -i -e 's/# CONFIG_CRYPTO is not set/CONFIG_CRYPTO=y/' .config
+	yes "" | make $makeflags HOSTCC="gcc -DPATH_MAX=4096" ARCH=$arch oldconfig
+	if make $makeflags CC="gcc-4.4 $ccflags" HOSTCC="gcc -DPATH_MAX=4096" \
+	     ARCH=$arch modules_prepare >Log-prepare 2>&1; then
+		date > $bintree/configured-$arch
+	else
+		cd $srctree
+		mv $bintree $bintree-bad
+	fi
+done
diff --git a/farm-config64 b/farm-config64
new file mode 100755
index 0000000..7e8b82d
--- /dev/null
+++ b/farm-config64
@@ -0,0 +1,32 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+arch=x86_64
+ccflags=-m64
+
+for srctree in $farm/*; do
+	test -d $srctree/Documentation || continue
+	bintree=$srctree-$arch
+	rm -rf $bintree
+	rm -rf $bintree-bad
+	mkdir $bintree
+	cd $srctree
+	make $makeflags HOSTCC="gcc -DPATH_MAX=4096" KBUILD_OUTPUT=$bintree \
+		CONFIG_WIRELESS_EXT=y CONFIG_NET_RADIO=y CONFIG_IPW2100=y \
+		CONFIG_CRYPTO=y ARCH=$arch defconfig
+	cd $bintree
+	sed -i -e 's/# CONFIG_NET_RADIO is not set/CONFIG_NET_RADIO=y/' .config
+	sed -i -e 's/# CONFIG_WIRELESS_EXT is not set/CONFIG_WIRELESS_EXT=y/' .config
+	sed -i -e 's/# CONFIG_IPW2100 is not set/CONFIG_IPW2100=y/' .config
+	sed -i -e 's/# CONFIG_CRYPTO is not set/CONFIG_CRYPTO=y/' .config
+	yes "" | make $makeflags HOSTCC="gcc -DPATH_MAX=4096" ARCH=$arch oldconfig
+	if make $makeflags CC="gcc-4.4 $ccflags" HOSTCC="gcc -DPATH_MAX=4096" \
+	     ARCH=$arch modules_prepare >Log-prepare 2>&1; then
+		date > $bintree/configured-$arch
+	else
+		cd $srctree
+		mv $bintree $bintree-bad
+	fi
+done
diff --git a/farm-hardlink b/farm-hardlink
new file mode 100755
index 0000000..11054d1
--- /dev/null
+++ b/farm-hardlink
@@ -0,0 +1,6 @@
+#! /bin/sh
+
+set -e
+source ~/.farmrc
+
+hardlink -t $farm
diff --git a/farm-warnings b/farm-warnings
new file mode 100755
index 0000000..83d959c
--- /dev/null
+++ b/farm-warnings
@@ -0,0 +1,3 @@
+#! /bin/sh
+
+grep -E -h -i '(error|warning):' Log* | grep -v /Module.symvers |sort -u
